src/renderer/components/+preferences/select-metrics-dialog.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { observer } from "mobx-react";
+import { action, observable } from "mobx";
+import { Dialog, DialogProps } from "../dialog/dialog";
+import { Checkbox } from "../checkbox/checkbox";
+import { Wizard, WizardStep } from "../wizard/wizard";
+import { userStore } from "../../../common/user-store";
+import { Select, SelectOption } from "../select/select";
+import { Icon } from "../icon/icon";
+
+interface Props extends Partial<DialogProps> {
+}
+
+export enum ResourceType {
+  Cluster = "Cluster",
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+@observer
+export class SelectMetricsDialog extends React.Component<Props> {
+
+  @observable static isOpen = false;
+
+  static open() {
+    SelectMetricsDialog.isOpen = true;
+  }
+
+  static close() {
+    SelectMetricsDialog.isOpen = false;
+  }
+
+  close = () => {
+    SelectMetricsDialog.close();
+  };
+
+  @action
+  onChange(value: ResourceType) {
+    userStore.hiddenMetrics.has(value) ? userStore.hiddenMetrics.delete(value) : userStore.hiddenMetrics.add(value);
+  }
+
+  reset = () => {
+    userStore.hiddenMetrics.clear();
+    userStore.preferences.hideMetrics = false;
+  };
+
+  render() {
+    const { ...dialogProps } = this.props;
+    let placeholder = <>Select metrics..</>;
+    const header = <h5>Select the metrics to hide</h5>;
+
+    if (userStore.hiddenMetrics.size >= 1) placeholder = <>Metrics: {Array.from(userStore.hiddenMetrics).join(", ")}</>;
+
+    return (
+      <Dialog
+        {...dialogProps}
+        isOpen={SelectMetricsDialog.isOpen}
+        close={this.close}
+      >
+        <Wizard header={header} done={this.close}>
+          <WizardStep contentClass="flow column" nextLabel={"Reset"} next={this.reset} prevLabel={"Close"}>
+            <div>
+              <Select
+                placeholder={placeholder}
+                isMulti
+                closeMenuOnSelect={false}
+                isDisabled={userStore.preferences.hideMetrics}
+                options={Object.values(ResourceType).map(res => res)}
+                onChange={([{ value }]: SelectOption[]) => {
+                  this.onChange(value);
+                }}
```

Comment:
I think it would be better if `this.onChange` accepted multiple options itself, so it could just be passed in here.

Reply:
Could you please elaborate a bit more? Do you mean in `onChange(resource: ResourceType[])`?

Reply:
Yeah that is what the signature should be

Reply:
yeah.. I tried it and unfortunately `Select` didn't work with `onChange(resource: ResourceType[]`
 I am confused about this moment.

Reply:
It seems I understand what happens. `([{ value }]: SelectOption[])` gets and passes only single element to `onChange(resource: ResourceType[])` func. That is why ` ResourceType[]` doesn't work. 
I changed inside `Select`
`           onChange={(value: SelectOption[]) => {
                  this.onChange(value.map(e => e.value));
                }}`
and `onChange(resource: ResourceType[])` works fine.